pkg/pool-stable-phantom/contracts/StablePhantomPool.sol

```diff
@@ -681,12 +696,13 @@ contract StablePhantomPool is
             _revert(Errors.UNHANDLED_EXIT_KIND);
         }
 
+        // Add amountsIn to get post-join balances
         // Subtract amountsOut to get post-exit balances
         // amountsOut returned from specific exit functions has BPT added back in for the Vault
         _mutateAmounts(balancesWithoutBpt, _dropBptItem(amountsOut), FixedPoint.sub);
 
         // Pass in the post-exit balances to reset the protocol fee basis.
-        _updateInvariantAfterJoinExit(currentAmp, balancesWithoutBpt);
+        _updateInvariantAfterJoinExit(currentAmp, balancesWithoutBpt, preJoinExitInvariant, virtualSupply, bptAmountIn);
```

Comment:
`_updateInvariantAfterJoinExit` assumes `bptAmountIn` is an amount out (minted), but here we're burning it.